src/OrchardCore.Themes/TheBlogTheme/Recipes/blog.recipe.json

```diff
@@ -1177,9 +1177,9 @@
         {
           "Source": "Sql",
           "Name": "RecentBlogPosts",
-          "Template": "SELECT DocumentId FROM ContentItemIndex WHERE ContentType='BlogPost' ORDER BY CreatedUtc DESC LIMIT 3",
+          "Template": "SELECT DocumentId FROM ContentItemIndex WHERE ContentType='BlogPost' and Published = 1 ORDER BY CreatedUtc DESC LIMIT 3",
```

Comment:
This one is puzzling me and I did not spend much time to look into it. Why did it work in the past? I mean there is a test that should have failed by was passing. The query should only return published blogposts

Reply:
I think there is a bug in the tests, it looks like they are running against Lucene only and never uses the Sql query...

Reply:
The `blog.recipe.json` add `RecentBlogPosts` index as `Sql` not as `Lucene`. I am not sure what is wrong here, but it looks like the test.

Reply:
The unit tests are always overwriting this with a Lucene query in
https://github.com/OrchardCMS/OrchardCore/blob/80ef1c7a0860a35b0260e833ee6b0161cd53e442/test/OrchardCore.Tests/Apis/Context/BlogContext.cs#L20
I noticed that a few days ago, but haven't created a bug report.